internal/core/src/index/Utils.cpp

```diff
@@ -273,4 +277,76 @@ AssembleIndexDatas(
     }
 }
 
+void
+AssembleIndexDataToDisk(
+    std::map<std::string, storage::FieldDataChannelPtr>& index_datas,
+    const std::filesystem::path& filepath) {
+    auto dir = filepath.parent_path();
+    std::filesystem::create_directories(dir);
+
+    int fd =
+        open(filepath.c_str(), O_CREAT | O_TRUNC | O_RDWR, S_IRUSR | S_IWUSR);
+    AssertInfo(fd != -1,
+               fmt::format("failed to create mmap file {}: {}",
+                           filepath.c_str(),
+                           strerror(errno)));
+
+    storage::FieldDataPtr data;
+    if (auto meta_iter = index_datas.find(INDEX_FILE_SLICE_META);
+        meta_iter != index_datas.end()) {
+        auto raw_metadata_array =
+            storage::CollectFieldDataChannel(meta_iter->second);
+        auto raw_metadata = storage::MergeFieldData(raw_metadata_array);
+        Config metadata = Config::parse(
+            std::string(static_cast<const char*>(raw_metadata->Data()),
+                        raw_metadata->Size()));
+
+        for (auto& item : metadata[META]) {
+            std::string prefix = item[NAME];
+            int slice_num = item[SLICE_NUM];
+            auto total_len = static_cast<size_t>(item[TOTAL_LEN]);
+
+            for (auto i = 0; i < slice_num; ++i) {
+                std::string file_name = GenSlicedFileName(prefix, i);
+                auto it = index_datas.find(file_name);
+                AssertInfo(it != index_datas.end(), "lost index slice data");
```

Comment:
How to ensure closing the file descriptor after an assertion interruption. 🤔

Reply:
Nice point, we should wrap this with a RAII type, to make sure it would be properly closed